### PR TITLE
adding summit stuff to ornl-ci

### PR DESCRIPTION
**Pull request by ketancmaheshwari**

### CODE REVIEW

**Comment by mtitov** [comment]
quick comments:

(1) Since we have two shared spaces, then we need to set that with the following variables (or set them within GitLab settings)

```
variables:
  SPACK_WORK_DIR_OPEN: /gpfs/wolf/csc449/proj-shared/sdk/
  SPACK_WORK_DIR_MODERATE: ...
```

(2) in `spack-setup`, `spack-update` need to add the following options (since it should be set in different spaces)

```
parallel:
  matrix:
    - HOST: "ascent"
      SPACK_WORK_DIR: ${SPACK_WORK_DIR_OPEN}
    - HOST: "summit"
      SPACK_WORK_DIR: ${SPACK_WORK_DIR_MODERATE}
tags:
  - $HOST
  - nobatch
```

(3) missed `summit` setup in `spack-build`

File: .gitlab/ornl-ci.yml [hunk 1]
```diff
@@ -31,6 +31,11 @@ env-setup:
     matrix:
       - HOST: [ "ascent" ]
         SPACK_ENV_NAME: [ "ascent" ]
+      - HOST: [ "summit" ]
+        SPACK_ENV_NAME: [ "rhel8-ppc64le" ]
```
Comment (mtitov) [inline .gitlab/ornl-ci.yml:35]: `SPACK_ENV_NAME: [ "summit" ]`

File: .gitlab/ornl-ci.yml [hunk 2]
```diff
@@ -31,6 +31,11 @@ env-setup:
...
+        COMPILER: [ "gcc@10.2.1" ]
+        TARGET_ARCH_OPT: [ "target=ppc64le" ]
```
Comment (mtitov) [inline .gitlab/ornl-ci.yml:37]: it is not used below, thus for now it could be removed (or extend the following commands with these parameters)

**Comment by ketancmaheshwari** [comment]
I made those changes in my latest commit.

**Comment by mturilli** [comment]
@mtitov is this ready to be merged?

**Pull request closed without merging**